src/components/search-date-picker.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import 'react-dates/initialize';
+import { DateRangePicker } from 'react-dates';
+import 'react-dates/lib/css/_datepicker.css';
+import moment from 'moment';
+import { isInclusivelyBeforeDay } from 'react-dates';
+
+/**
+ * @typedef SearchDatePicker
+ * @property {Function} onDateSearch
+ */
+
+/**
+ * Renders date picker which allows startDate, endDate or both to be selected
+ * Once a date is selected, onDateSearch function is called with startDate and endDate values
+ * Dates can also be cleared out after selected
+ *
+ * @class SearchDatePicker
+ * @extends {React.Component}
+ * @param {SearchDatePickerProps} props
+ */
+export default class SearchDatePicker extends React.Component {
```

Comment:
Hmmmm, I’m worried I might be missing the obvious, but is this component doing anything useful other than handling the focus state and providing some simple defaults?

If so, you could probably remove the dates from the state and simplify quite a bit:

```js
export default class SearchDatePicker extends React.Component {
  constructor (props) {
    super(props);
    this.state = { focusedInput: null };
  }

  // No componentDidUpdate or onDatesChange necessary.

  render() {
    return (
      <DateRangePicker
        startDateId="startDate"
        endDateId="endDate"
        startDate={this.props.startDate}
        endDate={this.props.endDate}
        onDatesChange={this.props.onDatesChange}
        focusedInput={this.state.focusedInput}
        onFocusChange={(focusedInput) => { this.setState({ focusedInput });}}
        isOutsideRange={day => !isInclusivelyBeforeDay(day, moment())}
        readOnly
        showClearDates />
    );
  }
}
```

This would require you to pass `startDate` and `endDate` in the props, but I’m pretty sure we’re goint to want to start doing that soon anyway (so we can restore the right value when switching between page list and details views).